python_modules/dagster/dagster/_core/definitions/auto_materialize_rule.py

```diff
@@ -660,19 +784,39 @@ def evaluate_for_asset(self, context: RuleEvaluationContext) -> RuleEvaluationRe
 
             nonexistent_parent_keys = {parent.asset_key for parent in nonexistent_parent_partitions}
             if nonexistent_parent_keys:
-                asset_partitions_by_nonexistent_but_required_parent_keys[
-                    frozenset(nonexistent_parent_keys)
+                asset_partitions_by_evaluation_data[
+                    WaitingOnAssetsRuleEvaluationData(frozenset(nonexistent_parent_keys))
                 ].add(candidate)
 
-        if asset_partitions_by_nonexistent_but_required_parent_keys:
-            return [
-                (
-                    WaitingOnAssetsRuleEvaluationData(waiting_on_asset_keys=k),
-                    v,
-                )
-                for k, v in asset_partitions_by_nonexistent_but_required_parent_keys.items()
-            ]
+        return self.add_evaluation_data_from_previous_tick(
+            context,
+            asset_partitions_by_evaluation_data,
+            should_use_past_data_fn=lambda ap: ap not in candidates_to_evaluate,
+        )
+
+
+@whitelist_for_serdes
+class DiscardOnMaxMaterializationsExceededRule(
```

Comment:
Is this a relic from a different PR?

Reply:
yeah this class ended up underneath the most recent skip rule, this is the correct place for it but I'll just move it back to make the diff nicer (I was planning on doing some other cleanup here anyway so I can do that in the next PR)